packages/forms/src/Components/Select.php

```diff
@@ -818,18 +821,15 @@ public function relationship(string | Closure | null $name = null, string | Clos
                     //
                     // https://github.com/filamentphp/filament/issues/1111
                     $relatedModels
-                        ->pluck($relationship->getRelatedKeyName())
+                        ->pluck($relationship->getRelated()->getKeyName())
```

Comment:
```suggestion
                        ->pluck(($relationship instanceof BelongsToMany) ? $relationship->getRelatedKeyName() : $relationship->getRelated()->getKeyName())
```

Reply:
@danharrin Out of curiosity, whats the difference here?

Reply:
check the implementation of getRelatedKeyName in BelongsToMany, it seems to differ, I think BelongsToMany has its own related key definition

Reply:
Cool, i thought `->getRelatedKeyName()` and `->getRelated()->getKeyName()` results would be the same